src/tools/tidy/src/main.rs

```diff
@@ -19,6 +19,11 @@ use std::path::{PathBuf, Path};
 use std::env;
 
 macro_rules! t {
+    ($e:expr, $p:expr) => (match $e {
+        Ok(e) => e,
+        Err(e) => panic!("{} failed on {} with {}", stringify!($e), ($p).display(), e),
+    });
```

Comment:
@alexcrichton the path name does not always show up elsewhere, or at least I couldn't find it; I added the path name to debug some problems I was seeing around symbolic links. I can remove it, but it seems a shame to not know what file is encountering a problem.